tasks/create-pyxis-image/create-pyxis-image.yaml

```diff
@@ -153,6 +157,14 @@ spec:
                   --rh-push $(params.rhPush) | tee "/tmp/output"
                 # The rh-push-to-external-registry e2e test depends on this line being in the task log
                 IMAGEID=$(awk '/The image id is/{print $NF}' /tmp/output)
+
+                # Remove the new image tags from all previous images
+                PYXIS_CERT_PATH=/tmp/crt PYXIS_KEY_PATH=/tmp/key cleanup_tags \
```

Comment:
@johnbieren I have a question for you: In this initial implementation, the cleanup_tags script is run once per pyxis container image. (At first I thought it could be incorporated in the create_image script, but having it separate seems nicer and the one extra get_image request that is needed this way is not too expensive.) So the script will run the same `find_repository_images_by_registry_path_tag` query multiple times - once per arch. And then filter out images with other archs. This `find_repository_images_by_registry_path_tag` request is quite expensive I'd say. So maybe add a "multiarch-mode" to the script which lets you list all the image ids for the different architectures at once and do the cleanup together? The query is still run for each tag, but if you have N architectures, it would run X/N times compared to now. Maybe it's worth it to do that? WDYT?

Here in the task, instead of running `cleanup_tags $IMAGEID` for each image, we would run something like `cleanup_tags $IMAGEIDARCH1 $IMAGEIDARCH2 $IMAGEIDARCH3 $IMAGEIDARCH4` and it would be after this inner while loop, once all the architectures for the component are saved in the json.

If it's not clear what I mean, let me know and we can discuss in offline.

Reply:
I am fine with the first approach. We can monitor the performance. Can you capture that idea in the ticket or something so it isn't lost?

Reply:
ok, I added a note about this to the jira.